### PR TITLE
Some more features for the number types.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: executor/src/witgen/affine_expression.rs [hunk 1]
```diff
@@ -231,13 +232,13 @@ where
         }
 
         // Check if they are mutually exclusive and compute assignments.
-        let mut covered_bits: <T as FieldElement>::Integer = 0u32.into();
+        let mut covered_bits: <T as FieldElement>::Integer = 0.into();
```
Comment (leonardoalt) [inline executor/src/witgen/affine_expression.rs:235]: `T::zero`?

File: executor/src/witgen/affine_expression.rs [hunk 2]
```diff
@@ -231,13 +232,13 @@ where
...
         let mut assignments = EvalValue::complete(vec![]);
         let mut offset = (-self.offset).to_integer();
         for (i, coeff, constraint) in parts {
             let constraint = constraint.clone().unwrap();
             let mask = constraint.mask();
-            if *mask & covered_bits != 0u32.into() {
+            if *mask & covered_bits != 0.into() {
```
Comment (leonardoalt) [inline executor/src/witgen/affine_expression.rs:241]: T::zero

File: executor/src/witgen/range_constraints.rs [hunk 3]
```diff
@@ -39,7 +41,7 @@ impl<T: FieldElement> RangeConstraint<T> {
 
     /// The range constraint of the sum of two expressions.
     pub fn try_combine_sum(&self, other: &RangeConstraint<T>) -> Option<RangeConstraint<T>> {
-        if self.mask & other.mask == 0u32.into() {
+        if self.mask & other.mask == 0.into() {
```
Comment (leonardoalt) [inline executor/src/witgen/range_constraints.rs:44]: T::zero, but also fine as is

**Comment by chriseth** [comment]
@leonardoalt updated

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**